Intro to React Post (#24)
datumbrain/datumbrain.github.io: _posts/2020-02-09-intro-to-react.md

---
layout: post
title: Introduction to React
author: Hafiz Muhammad Bilal
authorUrl: https://github.com/HafizMBilal009
date: "2020-02-09 12:40:31"
---

### React is a very popular **JavaScript** library for building user interfaces

## Pre-requisites

> Basic knowledge of :

Reviewer: ```suggestion
Basic knowledge of :
```

> Basic knowledge of :

Reviewer: Remove the line.

- HTML
- Javascript

## Why React?

> React was introduced by Facebook in 2013. Due to its readable syntax & code reusability, It has become one of the most popular & most used libraries for front-end development.

Reviewer: ```suggestion
React was introduced by Facebook in 2013. Due to its readable syntax & code re-usability, it has become one of the most popular & most used libraries for front-end development.
```

React has a very strong community on the internet world so most issues during development are solved quickly.

> React is easy to learn as its JSX (Javascript XML) syntax is very close to HTML and enhances the confidence of a novice developer.

Reviewer: The bullets should not be ended with a full-stop.

## Basic Concepts of React

> We will discuss some basic fundamental concepts of React :

Reviewer: ```suggestion
We will discuss some basic fundamental concepts of React:
```

### 1. Components

> Components are building blocks of a React app. A component has one or more than one element of HTML to re-use it wherever needed. With the use of components, we get rid of writing redundant code. There are two types of components i.e Class Components and Functional Components

Reviewer: Class Components and Functional Components can be put in bullets here.

### 2. JSX

> JSX is a way to write React code. As from it's name we can understand it's a mixture of Javascript and XML as a result of a syntax which is easy to understand.

Reviewer: ```suggestion
JSX is a way to write React code. As from its name we can understand it's a mixture of JavaScript and XML as a result of a syntax which is easy to understand.
```

> For example :

Reviewer: ```suggestion
For example:
```

```jsx
const element = <h1>Welcome to React World!</h1>;
```

### 3. Props

> Props are arguments we can pass into React components. Basically props are used for connecting different components by passing some as an attribute to a React component.

Reviewer: ```suggestion
Props are arguments we can pass into React components. Basically props are used for connecting different components by passing some value as an attribute to a React component.
```

> One component passes a value to the other one and other component can use this value in it.

Reviewer: ```suggestion
One component can pass value in the other.
```

### 4. State

State is the heart of a React app. Basically, state is used to store data which is required in our React component. Using state, changes in data can be seen in real-time without refreshing webpage.

## Enough Talk, Let's Code

We will build a simple React app which shows a list of quotes with their authors. We can add a quote and see a live update in the quotes table.

> Let's start by creating a simple React app by running this command :

Reviewer: ```suggestion
Let's start by creating a simple React app by running this command:
```

  };
  return (
    <div className="add-quote-form">
      <div>
        <label>Quote</label>
        <input
          type="text"
          value={quote}
          onChange={(e) => setQuote(e.target.value)}
        />
      </div>
      <div>
        <label>Author</label>
        <input
          type="text"
          value={author}
          onChange={(e) => setAuthor(e.target.value)}
        />
      </div>
      <button onClick={handleClick}>ADD</button>
    </div>
  );
}
export default AddQuoteForm;
```

This component will create a form to take quote and author as inputs and on button click, quote will be appended to the quotes table. This update will be real-time.
This is the beauty of React!
Remove everything from `App.css` and add the following code for better representation.

```css
table,
th,
td {
  border: 1px solid black;
  border-collapse: collapse;
}
th,
td {
  padding: 1rem;
}
table {
  border-spacing: 5px;
  width: 50%;
  margin-left: 25%;
  margin-top: 1rem;
}
.add-quote-form {
  width: 50%;
  margin-left: 25%;
  margin-top: 1.5rem;
}
.add-quote-form input {
  width: 100%;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
  height: 1.5rem;
}
```

> You can get this code on [Github](https://github.com/HafizMBilal009/my-app.git)

Reviewer: ```suggestion
[Here](https://github.com/HafizMBilal009/my-app.git) is the code repository you can use following this blog.
```
